tests/python/test_pointwise.py

```diff
@@ -65,3 +65,19 @@ def create_fusion(fd: FusionDefinition) -> None:
             torch.logical_or(ins[0] == 1, ins[2].unsqueeze(-1) == 1), 0, ins[1]
         ),
     )
+
+
+# Tests that CPU scalar tensor can be instantiated using fd.from_pytorch
+def test_cpu_add():
+    inputs = [torch.tensor(2.0, device="cpu", dtype=torch.float)]
+
+    def fusion_func(fd: FusionDefinition):
+        t0 = fd.from_pytorch(inputs[0])
+        s0 = fd.define_scalar(3.0)
+        t1 = fd.ops.add(t0, s0)
+        fd.add_output(t1)
+
+    with FusionDefinition() as fd:
+        fusion_func(fd)
+    nvf_out = fd.execute(inputs)
+    torch.testing.assert_close(nvf_out[0].cpu(), inputs[0] + 3.0)
```

Comment:
why do we need to call `cpu()` on nvf_out[0]. Are we producing output on cuda device by default? that doesn't feel right here.

Reply:
Yes, it is a cuda tensor.
Through the codegen path, we do not generate CPU scalar tensors so I would expect this.

Reply:
sorry this falls off my radar earlier. Thanks for requesting review again. :bow: 

So this is supposed to be an error coming from nvfuser I think, since we are being wild on producing output Tensor on an arbitrary device. Can you open an issue with this test as a repro, when the PR is merged?

Meanwhile, for this test, I suggest that we modify the test as a scalar cpu tensor add a gpu non-scalar tensor. Which should produce an output on gpu as expected.

Reply:
> So this is supposed to be an error coming from nvfuser I think, since we are being wild on producing output Tensor on an arbitrary device. Can you open an issue with this test as a repro, when the PR is merged?

Why should this be an error instead of on CPU? Maybe we should be using the expression evaluator here?

Reply:
Yeah, if we can generate a tensor on cpu that'll be great.

By throwing an error I was just referring to `we should throw an error if nvfuser generate an output tensor on gpu **instead** of cpu`